src/shogun/kernel/PolyKernel.cpp

```diff
@@ -71,8 +72,16 @@ void CPolyKernel::init()
 	m_gamma = 1.0;
 
 	set_normalizer(new CSqrtDiagKernelNormalizer());
-	SG_ADD(&degree, "degree", "Degree of polynomial kernel", ParameterProperties::HYPER);
-	SG_ADD(&m_c, "c", "The kernel is inhomogeneous if the value is higher than 0", ParameterProperties::HYPER);
-	SG_ADD(&m_gamma, "gamma", "Scaler for the dot product", ParameterProperties::HYPER);
+	SG_ADD(
+	    &degree, "degree", "Degree of polynomial kernel",
+	    ParameterProperties::HYPER);
+	SG_ADD(
+	    &m_c, "c", "The kernel is inhomogeneous if the value is higher than 0",
+	    ParameterProperties::HYPER);
+	SG_ADD(
+	    &m_gamma, "gamma", "Scaler for the dot product",
+	    ParameterProperties::HYPER | ParameterProperties::AUTO,
+	    [num_lhs = &num_lhs]() {
```

Comment:
this should be `lhs->get_dim_feature_space()`

Reply:
ahh yes that makes sense

Reply:
I had to capture `this` by value, instead of `lhs` by reference.. Casting `lhs` to `CDotFeatures*` and accessing `get_dim_feature_space()` led to a dangling pointer somehow... Do you know what the issue could be? Basically doing this:
```
[&lhs=lhs]() {
  return make_any(
     1.0 / static_cast<double>(((CDotFeatures*)lhs)->get_dim_feature_space()));
});
```